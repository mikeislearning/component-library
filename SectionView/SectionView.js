import React from 'react';
import { View, Text } from 'react-native';
import { PropTypes } from 'prop-types';

import styles from './SectionView.styles.js';

export default class SectionView extends React.Component {
  static propTypes = {
    sectionHeader: PropTypes.string.isRequired,
    children: PropTypes.node,
  };

  render() {
    const { sectionHeader, children } = this.props;

    return (
      <View style={styles.sectionContainer}>
        <Text style={styles.sectionHeader}>{sectionHeader}</Text>
        {children}
      </View>
    );
  }
}
