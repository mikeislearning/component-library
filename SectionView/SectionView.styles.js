import { StyleSheet } from 'react-native';
import { fonts } from 'app/theme';

export default StyleSheet.create({
  sectionContainer: {
    paddingHorizontal: 20,
    paddingVertical: 20,
    position: 'relative',
  },
  sectionHeader: {
    fontSize: fonts.size.medium,
    fontFamily: fonts.family.messinaSans.regular,
  },
});
