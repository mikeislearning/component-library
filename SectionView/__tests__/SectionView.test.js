/**
 * @format
 */

import React from 'react';
import { render } from 'react-native-testing-library';

import SectionView from '../SectionView';

describe('<SectionView />', () => {
  it('renders correctly', () => {
    const sectionView = render(<SectionView sectionHeader={'sectionHeader'} />);
    expect(sectionView).toMatchSnapshot();
    const json = sectionView.toJSON();
    const { children } = json;
    const [first] = children;
    expect(first.type).toEqual('Text');
    expect(first.children[0]).toEqual('sectionHeader');
  });
});
