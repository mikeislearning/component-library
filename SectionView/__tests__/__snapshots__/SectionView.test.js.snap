// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionView /> renders correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 20,
      "paddingVertical": 20,
      "position": "relative",
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "MessinaSans-Regular",
        "fontSize": 14,
      }
    }
  >
    sectionHeader
  </Text>
</View>
`;
